### DISCUSSION TITLE
[OP Stack Deployment] Generate contract artifacts

### DISCUSSION THREAD

**Discussion by k0de26** [ethereum-optimism/developers#295] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi guys, I'm following Creating Your Own L2 Rollup Testnet Tutorial and in the [Deploy the L1 contracts section](https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#deploy-the-l1-contracts) when running the `forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_URL` in the section of Generate Contract Artifacts it seems that most of the the script was executed correctly but not for SafeProxyFactory, SafeSingleton, SystemOwnerSafe. See below and in the Logs.

1. `Deploy Tx not found for SafeProxyFactory skipping deployment artifact generation`

2. `Deploy Tx not found for SafeSingleton skipping deployment artifact generation`

3. `Deploy Tx not found for SystemOwnerSafe skipping deployment artifact generation  Synced temp deploy files, deleting /home/k0de/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy`

### Logs

`k0de@k0de:~/optimism/packages/contracts-bedrock$ forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_URL
[⠢] Compiling...
[⠆] Compiling 1 files with 0.5.17
[⠰] Compiling 135 files with 0.8.19
[⠑] Compiling 331 files with 0.8.15
[⠊] Solc 0.5.17 finished in 167.28ms
[⠆] Solc 0.8.19 finished in 42.81s
[⠔] Solc 0.8.15 finished in 1102.28s
Compiler run successful!
Script ran successfully.

== Logs ==
  Connected to network with chainid 11155420
  Storing temp deployment data in /home/k0de/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy
  DeployConfig: reading file /home/k0de/optimism/packages/contracts-bedrock/deploy-config/getting-started.json
  Deploying from Deploy
  Deployment context: getting-started
  Syncing 24 deployments
  Using deployment artifact /home/k0de/optimism/packages/contracts-bedrock/broadcast/Deploy.s.sol/11155420/run-latest.json
  Syncing deployment AddressManager: contract AddressManager
  Syncing deployment DisputeGameFactoryProxy: contract Proxy
  Syncing deployment L1CrossDomainMessenger: contract L1CrossDomainMessenger
  Syncing deployment L1CrossDomainMessengerProxy: contract ResolvedDelegateProxy
  Syncing deployment L1ERC721Bridge: contract L1ERC721Bridge
  Syncing deployment L1ERC721BridgeProxy: contract Proxy
  Syncing deployment L1StandardBridge: contract L1StandardBridge
  Syncing deployment L1StandardBridgeProxy: contract L1ChugSplashProxy
  Syncing deployment L2OutputOracle: contract L2OutputOracle
  Syncing deployment L2OutputOracleProxy: contract Proxy
  Syncing deployment OptimismMintableERC20Factory: contract OptimismMintableERC20Factory
  Syncing deployment OptimismMintableERC20FactoryProxy: contract Proxy
  Syncing deployment OptimismPortal: contract OptimismPortal
  Syncing deployment OptimismPortalProxy: contract Proxy
  Syncing deployment ProtocolVersions: contract ProtocolVersions
  Syncing deployment ProtocolVersionsProxy: contract Proxy
  Syncing deployment ProxyAdmin: contract ProxyAdmin
  Deploy Tx not found for SafeProxyFactory skipping deployment artifact generation
  Deploy Tx not found for SafeSingleton skipping deployment artifact generation
  Syncing deployment SuperchainConfig: contract SuperchainConfig
  Syncing deployment SuperchainConfigProxy: contract Proxy
  Syncing deployment SystemConfig: contract SystemConfig
  Syncing deployment SystemConfigProxy: contract Proxy
  Deploy Tx not found for SystemOwnerSafe skipping deployment artifact generation  Synced temp deploy files, deleting /home/k0de/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy`

### Additional Information

I need some help on how to get `SafeProxyFactory, SafeSingleton, SystemOwnerSafe` Syncing. Where should I look and how to fix this issue?

**Comment by k0de26** [comment]
I don't know what is going on here, but anyhow I got my OP Stack Rollup running. 🎉

I'll try to figure this out another time. 🤞

**Reply by sbvegan** [reply, marked as answer]
I'd start by taking a look at the deploy script, it appears that these are optional additions.

**Reply by k0de26** [reply]
Huh, will do. Thanks

**Comment by cristianberbecaru** [comment]
I got the same issue right now. Could you advise what fixed for you?
